From 6efcddcd9c8d01af9e640c66d3c071917a8f471e Mon Sep 17 00:00:00 2001
From: "Reynaldo H. Verdejo Pinochet" <reynaldo@collabora.com>
Date: Tue, 14 May 2013 11:47:14 -0400
Subject: [PATCH 1/3] Preliminary 'make it build' Android port

No functionality implemented but:

- disable configtime check for UUID on Android
- Avoid relying on uuid.h (Not present in the Android NDK)
---
 configure.ac             |    4 ++++
 libgupnp/gupnp-service.c |    8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/configure.ac b/configure.ac
index 6ff0393..efd0d19 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,6 +43,10 @@ case "$target_os" in
         AC_MSG_RESULT(Darwin)
         UUID=
         ;;
+     linux-android*)
+        AC_MSG_RESULT(Android)
+        UUID=
+        ;;
      *)
         AC_MSG_RESULT($target_os)
         UUID=uuid
diff --git a/libgupnp/gupnp-service.c b/libgupnp/gupnp-service.c
index 2c4fd45..7436ea2 100644
--- a/libgupnp/gupnp-service.c
+++ b/libgupnp/gupnp-service.c
@@ -43,6 +43,7 @@
 
 #ifdef G_OS_WIN32
 #include <rpc.h>
+#elif __BIONIC__
 #else
 #include <uuid/uuid.h>
 #endif
@@ -1097,6 +1098,13 @@ generate_sid (void)
         }
 
         return ret;
+#elif __BIONIC__
+        char *ret = NULL;
+
+        /* FIXME: Implement this */
+        ret = g_strdup_printf ("uuid:thisisatemporaryplaceholderforarealone");
+
+        return ret;
 #else
         uuid_t id;
         char out[39];
-- 
1.7.10.4

